src/GitHub.TeamFoundation.14/Services/VSGitExt.cs

```diff
@@ -13,24 +14,55 @@ namespace GitHub.VisualStudio.Base
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class VSGitExt : IVSGitExt
     {
+        IGitHubServiceProvider serviceProvider;
+        IVSUIContext context;
         IGitExt gitService;
 
-        public void Refresh(IServiceProvider serviceProvider)
+        [ImportingConstructor]
+        public VSGitExt(IGitHubServiceProvider serviceProvider)
         {
-            if (gitService != null)
-                gitService.PropertyChanged -= CheckAndUpdate;
-            gitService = serviceProvider.GetServiceSafe<IGitExt>();
-            if (gitService != null)
-                gitService.PropertyChanged += CheckAndUpdate;
+            this.serviceProvider = serviceProvider;
+            var factory = serviceProvider.GetService<IVSUIContextFactory>();
```

Comment:
Is there a reason not to import this via a constructor parameter?

Reply:
Not any more! Previously two MEF services were being exported. Fixed! 😄